test/src/main/java/org/apache/accumulo/test/upgrade/GCUpgrade9to10TestIT.java

```diff
@@ -118,38 +113,70 @@ public void gcUpgradeNoDeletesIT() throws Exception {
   }
 
   /**
-   * This is really hard to make happen - the minicluster can only use so little memory to start up.
-   * The {@link org.apache.accumulo.master.upgrade.Upgrader9to10} CANDIDATE_MEMORY_PERCENTAGE can be
-   * adjusted.
+   * Ensure that the size of the candidates exceeds the {@link Upgrader9to10}'s CANDIDATE_BATCH_SIZE
+   * and will clean up candidates in multiple batches, without running out of memory.
    */
   @Test
   public void gcUpgradeOutofMemoryTest() throws Exception {
     killMacGc(); // we do not want anything deleted
 
-    int somebignumber = 100000;
-    String longpathname = "aaaaaaaaaabbbbbbbbbbccccccccccddddddddddeeeeeeeeee"
-        + "ffffffffffgggggggggghhhhhhhhhhiiiiiiiiiijjjjjjjjjj"
-        + "kkkkkkkkkkkkkkkkkklllllllllllllllllllllmmmmmmmmmmmmmmmmmnnnnnnnnnnnnnnnn";
-    longpathname += longpathname; // make it even longer
+    int numberOfEntries = 100_000;
+    String longpathname = StringUtils.repeat("abcde", 100);
+    assertEquals(500, longpathname.length());
+
+    // dynamically get the batch size from the other class
+    Field CANDIDATE_BATCH_SIZE_field = Upgrader9to10.class.getDeclaredField("CANDIDATE_BATCH_SIZE");
+    CANDIDATE_BATCH_SIZE_field.setAccessible(true);
+    long CANDIDATE_BATCH_SIZE = CANDIDATE_BATCH_SIZE_field.getLong(null);
+    // sanity check to ensure that any batch size assumptions are still valid in this test
+    assertEquals(4_000_000, CANDIDATE_BATCH_SIZE);
+
+    // ensure test quality by making sure we have enough candidates to
+    // exceed the batch size at least ten times
+    long numBatches = numberOfEntries * longpathname.length() / CANDIDATE_BATCH_SIZE;
+    assertTrue("Expected numBatches between 10 and 15, but was " + numBatches,
+        numBatches > 10 && numBatches < 15);
+
     Ample.DataLevel level = Ample.DataLevel.USER;
 
     log.info("Filling metadata table with lots of bogus delete flags");
     try (AccumuloClient c = Accumulo.newClient().from(getClientProperties()).build()) {
-      addEntries(c, level.metaTable(), somebignumber, longpathname);
+      Map<String,String> expected = addEntries(c, level.metaTable(), numberOfEntries, longpathname);
+      assertEquals(numberOfEntries + numberOfEntries / 10, expected.size());
+
+      Range range = MetadataSchema.DeletesSection.getRange();
 
       sleepUninterruptibly(1, TimeUnit.SECONDS);
+      try (Scanner scanner = c.createScanner(level.metaTable(), Authorizations.EMPTY)) {
+        Map<String,String> actualOldStyle = new HashMap<>();
+        scanner.setRange(range);
+        scanner.forEach(entry -> {
+          String strKey = entry.getKey().getRow().toString();
+          assertFalse(expected.containsKey(strKey));
+          String strValue = entry.getValue().toString();
+          actualOldStyle.put(strKey, strValue);
+        });
+        assertEquals(expected.size(), actualOldStyle.size());
+        assertNotEquals(expected, actualOldStyle);
```

Comment:
Could check that the key sets are disjoint using `Collections.disjoint()`.

Reply:
Will do. That's definitely better.